velox/type/Conversions.h

```diff
@@ -407,6 +407,10 @@ struct Converter<TypeKind::VARCHAR, void, TRUNCATE, LEGACY_CAST> {
   static std::string cast(const Timestamp& val) {
     TimestampToStringOptions options;
     options.precision = TimestampToStringOptions::Precision::kMilliseconds;
+    if constexpr (!LEGACY_CAST) {
```

Comment:
It doesn't look like this usage of the template parameter is beneficial. Is it?

Reply:
It saves a condition check per row but not much more I guess, if compared with passing a function arg? The second change is scientific notation. We can see more then.

We could do refactoring afterwards and compare with adding a function arg in terms of perf. Even for the existing TRUNCATE template parameter, I think it only saves one condition check per row.

Reply:
@gggrace14 Makes sense. Let's make sure to prioritize this work and remove unnecessary template parameters. Templates are great, but they cause a significant increase in the binary size.

Reply:
Let's add a TODO here.

Reply:
Agree. Got you, Masha! I'll test refactoring after the scientific notation change.